### PR TITLE
minor updates for FV3 "the dycore" and run templates, parm/field_table/* updated for RTS

### LINKED ISSUE #411: need to update atmos_cubed_shere hash
## Description
Need to update the hash for atmos_cubed_sphere to bring in latest bugfixes and enhancements.

## Solution
updated hash

## Testing:
New hash has been tested in the ufs-weather-model RTS on gaea with success
Baseline RTS results are unchanged
Updates have also been tested in a future update by @XiaqiongZhou-NOAA

## Dependent PRs:
- associated ufs-weather-model/pull/870

### LINKED ISSUE #869: updated run templates to use field_table entries in tests/parm/field_table
## Description
continuing the work started by @junwang-noaa to move field_tables into tests/parm/field_table and update the run templates to ensure there are no collisions.

This is needed to support an fv3atm PR

## Solution
use new framework for bringing in field_tables from repo managed location

## Alternatives
N/A

## Related to
Directly reference any issues or PRs in this or other repositories that this is related to, and describe how they are related. Example:
- needed by noaa-emc/fv3atm/pull/412

### PR DESCRIPTION
**Pull request by bensonr**
# PR Checklist

- [x] Ths PR is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR. Please consult the ufs-weather-model [wiki](https://github.com/ufs-community/ufs-weather-model/wiki/Making-code-changes-in-the-UFS-weather-model-and-its-subcomponents) if you are unsure how to do this.

- [x] This PR has been tested using a branch which is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR

- [x] An Issue describing the work contained in this PR has been created either in the subcomponent(s) or in the ufs-weather-model. The Issue should be created in the repository that is most relevant to the changes in contained in the PR. The Issue and the dependent sub-component PR 
are specified below.

- [x] If new or updated input data is required by this PR, it is clearly stated in the text of the PR.

## Description

This PR brings in [RTS updates](https://github.com/junwang-noaa/ufs-weather-model/tree/field_table) from @junwang-noaa to fully use the FIELD_TABLE macro.  In addition the field_tables in tests/parm/field_table/ have been modified to provide correct initialization values.  Some of the run templates in tests/fv3_conf have been updated to reflect the FIELD_TABLE changes.  FInally, minor bugfixes and updates for FV3 "the dycore" are included to coincide with the field_table updates.

### Issue(s) addressed

Link the issues to be closed with this PR, whether in this repository, or in another repository.
(Remember, issues must always be created before starting work on a PR branch!) 
- fixes #869
- fixes noaa-emc/fv3atm/issues/411

## Testing

How were these changes tested? What compilers / HPCs was it tested with? Are the changes covered by regression tests? (If not, why? Do new tests need to be added?) Have regression tests and unit tests (utests) been run? On which platforms and with which compilers? (Note that unit tests can only be run on tier-1 platforms)

- [x] hera.intel
- [x] hera.gnu
- [x] orion.intel
- [x] cheyenne.intel 
- [x] cheyenne.gnu
- [x] gaea.intel 
- [x] jet.intel
- [x] wcoss_cray
- [x] wcoss_dell_p3
- [ ] CI. skip due to DATM submodule related changes in CI

## Dependencies

If testing this branch requires non-default branches in other repositories, list them. Those branches should have matching names (ideally).

Do PRs in upstream repositories need to be merged first?
If so add the "waiting for other repos" label and list the upstream PRs
- waiting on noaa-emc/fv3atm/pull/412

### CODE REVIEW

File: tests/fv3_conf/hafs_regional_run.IN [hunk 1]
```diff
@@ -1,4 +1,10 @@
 rm -rf INPUT RESTART
 mkdir  INPUT RESTART
 
-rsync -arv @[INPUTDATA_ROOT]/FV3_hafs_regional_input_data/. .
+rsync -arv --exclude 'field_table' @[INPUTDATA_ROOT]/FV3_hafs_regional_input_data/. .
+
+# overwrite Field table until the field_table in FV3_hafs_regional_input_data is removed
```
Comment (junwang-noaa) [inline tests/fv3_conf/hafs_regional_run.IN]: Can we remove the if statement here?

Comment (bensonr) [reply tests/fv3_conf/hafs_regional_run.IN]: I think it can, but I'll test it to be certain.

Comment (bensonr) [reply tests/fv3_conf/hafs_regional_run.IN]: tests for hafs_regional* pass on gaea

**Comment by junwang-noaa** [comment]
@bensonr would you please sync your ufs-weather-model branch with the latest develop branch and run RT on gaea? After that we will run RT on other platforms. Thanks

**Comment by MinsukJi-NOAA** [comment]
@bensonr please make these changes to tests/ci/repo_check.sh to address CI related issue. CI needs to be skipped for this PR since this change will not take effect until the next PR.
```
diff --git a/tests/ci/repo_check.sh b/tests/ci/repo_check.sh
index c74182c..843e004 100755
--- a/tests/ci/repo_check.sh
+++ b/tests/ci/repo_check.sh
@@ -13,8 +13,8 @@ result() {
 }
 
 # Declare variables
-declare -A base fv3 mom6 cice ww3 stoch gocart nems cmeps cdeps cmake
-submodules="fv3 mom6 cice ww3 stoch gocart nems cmeps cdeps cmake"
+declare -A base fv3 mom6 cice ww3 stoch gocart nems cmeps cdeps hycom cmake
+submodules="fv3 mom6 cice ww3 stoch gocart nems cmeps cdeps hycom cmake"
 comment=''
 ownerID=$1
 
@@ -60,7 +60,7 @@ cdeps[branch]='develop'
 cdeps[dir]='CDEPS-interface/CDEPS'
 
 hycom[repo]='https://github.com/NOAA-EMC/HYCOM-src'
-hycom[branch]='develop'
+hycom[branch]='emc/develop'
 hycom[dir]='HYCOM-interface/HYCOM'
 
 cmake[repo]='https://github.com/NOAA-EMC/CMakeModules'

```

**Comment by BrianCurtis-NOAA** [comment]
***Automated RT Failure Notification***
Machine: cheyenne
Compiler: gnu
Job: RT
Repo location: /glade/scratch/dtcufsrt/autort/tests/auto/pr/759311817/20211018124512/ufs-weather-model
Please manually delete: /glade/scratch/dtcufsrt/FV3_RT/rt_30879
Test control 001 failed failed 
Test control 001 failed in run_test failed 
Please make changes and add the following label back:
cheyenne-gnu-RT

**Comment by BrianCurtis-NOAA** [comment]
***Automated RT Failure Notification***
Machine: hera
Compiler: intel
Job: RT
Repo location: /scratch1/NCEPDEV/nems/emc.nemspara/autort/pr/759311817/20211018183012/ufs-weather-model
Please manually delete: /scratch1/NCEPDEV/stmp2/emc.nemspara/FV3_RT/rt_27255
Test control_atm_aerosols 092 failed in check_result failed 
Test control_atm_aerosols 092 failed in run_test failed 
Test control_thompson_extdiag_debug 066 failed failed 
Test control_thompson_extdiag_debug 066 failed in run_test failed 
Test control_c384gdas_wav 091 failed failed 
Test control_c384gdas_wav 091 failed in run_test failed 
Please make changes and add the following label back:
hera-intel-RT

**Comment by climbfuji** [comment]
> /glade/scratch/dtcufsrt/autort/tests/auto/pr/759311817/20211018124512/ufs-weather-model

Not sure if this test timed out or not. It looks like the model ran to completion, but the post task was still doing something when the run aborted. Will recover the log file, rerun the failed tests and upload if successful.

**Comment by BrianCurtis-NOAA** [comment]
***Automated RT Failure Notification***
Machine: jet
Compiler: intel
Job: RT
Repo location: /lfs4/HFIP/h-nems/emc.nemspara/autort/pr/759311817/20211018183014/ufs-weather-model
Please manually delete: /lfs4/HFIP/h-nems/emc.nemspara/RT_RUNDIRS/emc.nemspara/FV3_RT/rt_263947
Test cpld_control_c192_p7 005 failed failed 
Test cpld_control_c192_p7 005 failed in run_test failed 
Please make changes and add the following label back:
jet-intel-RT

**Comment by climbfuji** [comment]
> > /glade/scratch/dtcufsrt/autort/tests/auto/pr/759311817/20211018124512/ufs-weather-model
> 
> Not sure if this test timed out or not. It looks like the model ran to completion, but the post task was still doing something when the run aborted. Will recover the log file, rerun the failed tests and upload if successful.

I pushed the full regression test log for Cheyenne/GNU.

**Comment by junwang-noaa** [comment]
Thanks, Dom

**Comment by DeniseWorthen** [comment]
jet was a time-out. I will rerun.

**Review by climbfuji — APPROVED** [review]
Nice cleanup work. I note the following:

1. We should do the same with `diag_table` - use version-controller files from the ufs-weather-model repository for the regression tests
2. When a new input data set is created in future pull requests, remove all `field_table` (and `diag_table` if (1) has been completed by then) from the new input data set.

**Comment by BrianCurtis-NOAA** [comment]
***Automated RT Failure Notification***
Machine: gaea
Compiler: intel
Job: RT
Repo location: /lustre/f2/pdata/ncep/emc.nemspara/autort/pr/759311817/20211018183019/ufs-weather-model
Please manually delete: /lustre/f2/scratch/emc.nemspara/FV3_RT/rt_27605
Test cpld_decomp_p7 004 failed failed 
Test cpld_decomp_p7 004 failed in run_test failed 
Please make changes and add the following label back:
gaea-intel-RT

**Comment by climbfuji** [comment]
> _**Automated RT Failure Notification**_
> Machine: gaea
> Compiler: intel
> Job: RT
> Repo location: /lustre/f2/pdata/ncep/emc.nemspara/autort/pr/759311817/20211018183019/ufs-weather-model
> Please manually delete: /lustre/f2/scratch/emc.nemspara/FV3_RT/rt_27605
> Test cpld_decomp_p7 004 failed failed
> Test cpld_decomp_p7 004 failed in run_test failed
> Please make changes and add the following label back:
> gaea-intel-RT

What happened here?

**Comment by DeniseWorthen** [comment]
Gaea was a time out for that one test. I was going to re-run.

**Comment by bensonr** [comment]
gaea is "sick" and been experiencing performance issues.  ORNL is planning to apply some fixes this week with the hope of addressing the Lustre issues that have been randomly plaguing the system for over two months.

**Comment by climbfuji** [comment]
> gaea is "sick" and been experiencing performance issues. ORNL is planning to apply some fixes this week with the hope of addressing the Lustre issues that have been randomly plaguing the system for over two months.

Time for gaea to get vaccinated ...

**Review by DusanJovic-NOAA — APPROVED** [review]

**Review by MinsukJi-NOAA — APPROVED** [review]

**Review by junwang-noaa — APPROVED** [review]

**Review by DeniseWorthen — APPROVED** [review]

**Comment by climbfuji** [comment]
fv3atm submodule pointer is correct, will merge.

**Pull request merged**